server.js

```diff
@@ -3107,6 +3107,100 @@ cache(function(data, match, sendBadge, request) {
   })}
 ));
 
+// DUB download integration
+camp.route(/^\/dub\/(dd|dw|dm|dt)\/([^\/]+)(?:\/([^\/]+))?\.(svg|png|gif|jpg|json)$/,
+cache(function (data, match, sendBadge, request) {
+  var info = match[1]; // downloads (dd - daily, dw - weekly, dm - monthly, dt - total)
+  var pkg = match[2]; // package name, e.g. vibe-d
+  var version = match[3]; // version (1.2.3 or latest)
+  var format = match[4];
+  var apiUrl = 'http://code.dlang.org/api/packages/'+pkg;
+  if (version)
+    apiUrl += '/' + version;
+  apiUrl += '/stats';
+  var badgeData = getBadgeData('dub', data);
+  request(apiUrl, function(err, res, buffer) {
+    if (err != null) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+    }
+    try {
+      var data = JSON.parse(buffer);
+      if (info.charAt(0) === 'd') {
+        badgeData.text[0] = getLabel('downloads', data);
+        var downloads;
+        switch (info.charAt(1)) {
+          case 'm':
+            downloads = data.downloads.monthly;
+            badgeData.text[1] = metric(downloads) + '/month';
+            break;
+          case 'w':
+            downloads = data.downloads.weekly;
+            badgeData.text[1] = metric(downloads) + '/week';
+            break;
+          case 'd':
+            downloads = data.downloads.daily;
+            badgeData.text[1] = metric(downloads) + '/day';
+            break;
+          case 't':
+            downloads = data.downloads.total;
+            badgeData.text[1] = metric(downloads);
+            break;
+        }
+        if (version)
+          badgeData.text[1] += ' ' + version;
```

Comment:
`badgeData.text[1] += ' ' + versionColor(version).version;` would be more fitting, to avoid having the download count number directly precede another number (having the `v` prefix the version numbers is usually a good indicator of what the data represents).